frontend/src/reducers/tables.js

```diff
@@ -107,6 +107,17 @@ const reducer = produce((draftState, action) => {
       return;
     }
 
+    case types.UPDATE_TABLE_SELECTION: {
+      const { windowId, viewId, selection } = action.payload;
+      const tableId = getTableId({ windowType: windowId, viewId });
+      draftState[tableId] = {
```

Comment:
`draftState` is already a copy of the object so it's easier to just set the `selected` property (like for `activeState`) , since we're not merging anything but setting it fresh.

Reply:
agree

Reply:
@petrican  any idea why?

Reply:
Yes. Setting the selection before the creation of the table data. This happens in the Table component. This I'm sure will be fixed along with the refactoring but as a precaution I will add a TODO in here also.